cogs/moderation.py

```diff
@@ -134,21 +139,58 @@ async def purge(self, ctx: commands.Context):
         """
         Commands for bulk deletion.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        if not ctx.invoked_subcommand:
+            await IBpyHelp.send_command_help(ctx)
     
     @purge.command()
-    async def message(self, ctx: commands.Context):
+    async def message(self, ctx: commands.Context, number: int):
         """
         Bulk delete messages.
         """ 
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        async for message in ctx.channel.history(limit=number):
+            try:
+                await message.delete()
+            except discord.Forbidden:
+                await ctx.reply('I do not have permission to delete messages.')
+                return
+            except discord.HTTPException:
+                await ctx.reply('An error occurred while deleting messages.')
+                return
+        await ctx.reply(f'Deleted {number} messages.', delete_after=5)
     
     @purge.command()
-    async def reaction(self, ctx: commands.Context):
+    async def reaction(self, ctx: commands.Context, channel: discord.TextChannel, message_id: int, emoji: str):
```

Comment:
Shouldn't `emoji` be some type of EmojiConverter? How would it work when a member provides an emote directly?